File: app/src/main/java/com/cere/csplayer/entity/Play.java
package com.cere.csplayer.entity;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.Nullable;

import com.litesuits.orm.db.annotation.Column;
import com.litesuits.orm.db.annotation.PrimaryKey;
import com.litesuits.orm.db.annotation.Table;
import com.litesuits.orm.db.enums.AssignType;

/**
 * Created by CheRevir on 2019/10/21
 */
@Table("play")
public class Play implements Parcelable {
    @PrimaryKey(AssignType.AUTO_INCREMENT)
    @Column("_id")
    private long id;
    @Column("position")
    private int position;
    
    public Play() {
    }

    protected Play(Parcel in) {
        position = in.readInt();
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    @Override
    public int hashCode() {
        return position;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (!(obj instanceof Play)) {
            return false;
        }
        return ((Play) obj).getPosition() == this.getPosition();
    }

    public static final Creator<Play> CREATOR = new Creator<Play>() {
        @Override
        public Play createFromParcel(Parcel in) {
            return new Play(in);
        }

        @Override
        public Play[] newArray(int size) {
            return new Play[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(position);
    }
}

File: app/src/main/java/com/cere/csplayer/entity/Play.kt
package com.cere.csplayer.entity

import android.os.Parcel
import android.os.Parcelable
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Created by CheRevir on 2020/9/5
 */
@Entity(tableName = "play")
data class Play(
    @ColumnInfo(name = "id") var id: Int,
    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "_id") var ID: Int = 0
) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.readInt(), parcel.readInt())

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other is Play) {
            return id == other.id
        }
        return false
    }

    override fun hashCode(): Int {
        return id
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(ID)
        parcel.writeInt(id)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun toString(): String {
        return "Play(_id=$ID, id=$id)"
    }

    companion object CREATOR : Parcelable.Creator<Play> {
        override fun createFromParcel(parcel: Parcel): Play {
            return Play(parcel)
        }

        override fun newArray(size: Int): Array<Play?> {
            return arrayOfNulls(size)
        }
    }
}